### PR TITLE
chore: improve default_load in certs.py

### PR DESCRIPTION
**Pull request by rb-determined-ai**
Previously, you were not able to do:

    client.login(
        master="https://localhost:8443",
        cert_name="self_signed_name",
    )

and still rely on the trust-on-first-use certificate store in
certs.json, because passing any explicit certificate-related args
prevented loading certs.json entirely.

The old logic could be summarized as:

 - If any explicit certificate args are set, use only explicit args.
 - If not, use the path at DET_MASTER_CERT_FILE.
 - If none, try loading from certs.json.

The new logic can be summarized as:

 - If noverify is set explicitly, or via DET_MASTER_CERT_FILE=noverify,
   disable verification entirely.
 - Otherwise, look for certificate contents in the following places:
    - the explicitliy configured path, if one is provided
    - at DET_MASTER_CERT_FILE, if that is set
    - in certs.json
 - Then look for a certificate name in the following places:
    - the explicitly configure name, if one is provided
    - as DET_MASTER_CERT_NAME, if that is set

The result is that the python sdk is a bit more usable when working with
self-signed certificates.

An alternative would be to save the validation hostname in certs.json at
the moment that you trust a certificate.

### CODE REVIEW

File: harness/determined/common/api/certs.py
Comment (ioga) [inline harness/determined/common/api/certs.py:223]: so if `explicit_path` doesn't exist, we'll error out in the `open` call. maybe we should also fail if `env_path` cert doesn't exist, instead of printing a warning? 
it'd be easier on the user if these are just different ways to set a config option, with the same treatment.

**Review by ioga — APPROVED** [review]

**Comment by rb-determined-ai** [comment]
After further discussion, we elected to not land this change at all.

**Pull request closed without merging**
